src/psyclone/domain/lfric/lfric_builtins.py

```diff
@@ -1555,8 +2100,26 @@ class LFRicIncXPowrealAKern(LFRicBuiltIn):
     ''' Raise a real-valued field to a real power and return it.
 
     '''
+    _case_name = "inc_X_powreal_a"
+    _datatype = "real"
+
+    @classmethod
+    def metadata(cls):
+        '''Returns the kernel metadata describing this builtin.
+
+        :returns: kernel metadata describing this builtin.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.LFRicKernelMetadata`
+
+        '''
+        gh_datatype = cls._datatype_lookup[cls._datatype]
+        return cls._builtin_metadata([
+            FieldArgMetadata(gh_datatype, "gh_readwrite", "any_space_1"),
+            ScalarArgMetadata(gh_datatype, "gh_read")])
+
     def __str__(self):
-        return "Built-in: Raise a real-valued field to a real power"
+        return (f"Built-in: {self._case_name} (Raise "
+                f"{a_or_an(self._datatype)} {self._datatype}-valued field "
+                f"to a real power)")
```

Comment:
I left this as the kernel is called `inc_X_powreal`, so I thought the `power` should always be `real`. However, I appreciate I changed the metadata so that the power would be an integer if we had an integer version, but that was actually my mistake (yet another one). So, my question to you is, what would an integer version of this built-in look like? I assumed the field could be an integer but not the power. Or perhaps there no integer equivalent? For the moment I've left it as it as and changed the 2nd meta_arg metadata to "gh_real".

Reply:
Specifying the power metadata explicitly as "gh_real" makes sense, thanks! The only `integer` equivalent I can think of that makes sense is with `integer` field *and* positive `integer` power, otherwise the result would be `real` and we already have built-ins for that. However, I am not sure we want to implement that as it would lead to overflow.